packages/commonwealth/server/migrations/20230217164706-root-id-refactor.js

```diff
@@ -0,0 +1,134 @@
+'use strict';
+
+// The purpose of this script is to turn root_id to thread_id. To achieve this without losing any data, we must set the
+// comments that pointed to a proposal, to instead point to a thread that is created to backlink to that proposal
+module.exports = {
+  up: async (queryInterface, Sequelize) => {
+    await queryInterface.sequelize.transaction(async (t) => {
+      const updateQueries = [];
+
+      const threadComments = await queryInterface.sequelize.query(
+        `SELECT * FROM "Comments" WHERE "root_id" ILIKE '%discussion%'`
+      );
+
+      // for each comment that is part of a discussion, replace its root_id with the thread_id
+      threadComments[0].forEach((c) => {
+        updateQueries.push(
+          queryInterface.sequelize.query(
+            `UPDATE "Comments" SET "root_id" = ${
+              c.root_id.split('_')[1]
+            } WHERE "id"=${c.id}`,
+            { transaction: t }
+          )
+        );
+      });
+
+      // we will now perform the same operation with proposal comments, but in order to do so we must create threads
+      // to backlink to proposals
+      const proposalComments = await queryInterface.sequelize.query(
+        `SELECT * FROM "Comments" WHERE "root_id" NOT ILIKE '%discussion%'`
+      );
+
+      // maps root_id -> newThread it creates
+      const newThreads = new Map();
+      proposalComments[0].forEach((c) => {
+        if (newThreads.get(c.root_id)) {
+          return; //if we already had created thread to backlink proposal comments, don't create another one
+        }
+
+        let rootParts = c.root_id.split('_');
+        // handle cosmosproposals differently
+        if (rootParts[0] === 'cosmosproposal') {
+          rootParts = ['proposal', rootParts[1]];
+        } else {
+          rootParts = [`proposal/${rootParts[0]}`, rootParts[1]];
+        }
+
+        // otherwise map the root_id to a new thread to backlink proposal comments
+        newThreads.set(c.root_id, {
+          address_id: 1,
+          title: `Thread for ${c.chain}'s ${rootParts[0]} ${rootParts[1]}`,
+          kind: 'link',
+          body: '',
+          url: `https://commonwealth.im/${c.chain}/${rootParts[0]}/${rootParts[1]}`,
+          chain: c.chain,
+          read_only: true,
```

Comment:
This was how I interpreted (from PRD):
> Then link this Archival Thread to Proposal (so comments are still “connected”)

But it may be wrong. I was thinking Archival threads would be like some deprecated read only thing. But if we decide to allow the user to comment on them, then I will change it.

Reply:
Ah, this should've been resolved -- I re-read the PRD and your version is correct.